rust/lance/src/io/writer/statistics.rs

```diff
@@ -0,0 +1,1482 @@
+    match arrays[0].data_type() {
+        DataType::Boolean => get_boolean_statistics(arrays),
+        DataType::Int8 => get_statistics::<Int8Type>(arrays),
+        DataType::UInt8 => get_statistics::<UInt8Type>(arrays),
+        DataType::Int16 => get_statistics::<Int16Type>(arrays),
+        DataType::UInt16 => get_statistics::<UInt16Type>(arrays),
+        DataType::Int32 => get_statistics::<Int32Type>(arrays),
+        DataType::UInt32 => get_statistics::<UInt32Type>(arrays),
+        DataType::Int64 => get_statistics::<Int64Type>(arrays),
+        DataType::UInt64 => get_statistics::<UInt64Type>(arrays),
+        DataType::Float32 => get_statistics::<Float32Type>(arrays),
+        DataType::Float64 => get_statistics::<Float64Type>(arrays),
+        DataType::Date32 => get_statistics::<Date32Type>(arrays),
+        DataType::Date64 => get_statistics::<Date64Type>(arrays),
+        DataType::Time32(TimeUnit::Second) => get_statistics::<Time32SecondType>(arrays),
+        DataType::Time32(TimeUnit::Millisecond) => get_statistics::<Time32MillisecondType>(arrays),
+        DataType::Time64(TimeUnit::Microsecond) => get_statistics::<Time64MicrosecondType>(arrays),
+        DataType::Time64(TimeUnit::Nanosecond) => get_statistics::<Time64NanosecondType>(arrays),
+        DataType::Timestamp(TimeUnit::Second, _) => get_statistics::<TimestampSecondType>(arrays),
+        DataType::Timestamp(TimeUnit::Millisecond, _) => {
+            get_statistics::<TimestampMillisecondType>(arrays)
+        }
+        DataType::Timestamp(TimeUnit::Microsecond, _) => {
+            get_statistics::<TimestampMicrosecondType>(arrays)
+        }
+        DataType::Timestamp(TimeUnit::Nanosecond, _) => {
+            get_statistics::<TimestampNanosecondType>(arrays)
+        }
+        DataType::Duration(TimeUnit::Second) => get_statistics::<DurationSecondType>(arrays),
+        DataType::Duration(TimeUnit::Millisecond) => {
+            get_statistics::<DurationMillisecondType>(arrays)
+        }
+        DataType::Duration(TimeUnit::Microsecond) => {
+            get_statistics::<DurationMicrosecondType>(arrays)
+        }
+        DataType::Duration(TimeUnit::Nanosecond) => {
+            get_statistics::<DurationNanosecondType>(arrays)
+        }
+        DataType::Decimal128(_, _) => get_decimal_statistics(arrays),
+        DataType::Binary => get_binary_statistics(arrays),
+        DataType::Utf8 => get_string_statistics(arrays),
+        DataType::LargeUtf8 => get_string_statistics(arrays),
+        DataType::Dictionary(_, _) => get_dictionary_statistics(arrays),
+        DataType::Struct(fields) => collect_struct_statistics(arrays, fields),
+        // TODO: List, LargeList, FixedSizeList, FixedSizeBinary, BinaryArray, Struct
+        _ => {
+            println!(
+                "Stats collection for {} is not supported yet",
+                arrays[0].data_type()
+            );
+            todo!()
+        }
+    }
+}
```

Comment:
@wjones127 I didn't really start with a plan for which types to cover. Do you think all of these should be covered: `List, LargeList, FixedSizeList, FixedSizeBinary, BinaryArray, Struct`? Should we remove some?

Reply:
FixedSizeBinary, BinaryArray, Struct: yes. By struct, you specifically mean children of struct, right? or do you mean something different?
List, LargeList, FixedSizeList: let's punt on those. It's unclear what the stats should be righ tnow.

Reply:
> By struct, you specifically mean children of struct, right?

Yeah and return them as a struct:
https://github.com/lancedb/lance/blob/5213999c01ffedfa1743aa4ef62e405edc6d36c1/rust/lance/src/io/writer/statistics.rs#L618-L640